cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1241,7 +1241,8 @@ def all_kube_system_pods_running():
     result = json.loads(output)
     for pod in result['items']:
         status = pod['status']['phase']
-        if status != 'Running':
+        # Evicted nodes should re-spawn
+        if status != 'Running' and pod['status']['reason'] != 'Evicted':
```

Comment:
Can we play it safe in case `'reason'` is not present? Something like `pod['status'].get('reason', '')`.